src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -252,6 +256,30 @@ describe('HeatmapPlot', () => {
     expect(component.find(Empty).length).toEqual(1);
   });
 
+  it('Shows loader if cell sets is empty', () => {
+    const store = mockStore({
+      ...initialState,
+      cellSets: {
+        ...initialState.cellSets,
+        hierarchy: [],
+        properties: [],
+        loading: false,
+        error: false,
+      },
+    });
+
+    component = mount(
+      <Provider store={store}>
+        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
+      </Provider>,
+    );
+
+    expect(component.find('VegaHeatmap').length).toEqual(0);
+
+    // 2 Because the Loader component contains ClipLoader, which contains a Loader
+    expect(component.find('Loader').length).toEqual(2);
```

Comment:
Agi, if you comment out the changes you did in `src/components/data-exploration/heatmap/HeatmapPlot.jsx`, would this test fail? Just to check if this test is catching this bug

Reply:
yes, the test fails

Reply:
It renders the heatmap instead of the loader

Reply:
ok, great.